packages/plugins/i18n/admin/src/contentManagerHooks/editView.tsx

```diff
@@ -124,4 +103,59 @@ const Span = styled(Flex)`
   }
 `;
 
+/**
+ * Compute effective localization for a specific field occurrence based on:
+ * - Content type i18n setting
+ * - Ancestor attribute (dynamic zone or component) i18n setting
+ * - Field-level i18n setting for root fields
+ */
+const getEffectiveLocalization = (
+  name: string,
+  attribute: EditFieldLayout['attribute'],
+  edit: EditLayout
+): boolean => {
```

Comment:
```suggestion
) => {
```
 
I'm not sure we need to explicitly type the return value here the boolean should be inferred